packages/@aws-cdk/aws-iam/lib/instance-profile.ts

```diff
@@ -0,0 +1,60 @@
+import { Construct, Token } from '@aws-cdk/cdk';
+import { cloudformation } from './iam.generated';
+import { Role } from './role';
+import { generateInstanceProfileName, undefinedIfEmpty } from './util';
+
+export interface InstanceProfileProps {
+
+  /**
+   * The path associated with this instance profile. For information about IAM Instance Profiles, see
+   * Friendly Names and Paths in IAM User Guide.
+   */
+  path?: string;
+
+  /**
+   * The name of an existing IAM role to associate with this instance profile.
+   * Currently, you can assign a maximum of one role to an instance profile.
+   */
+  roles: Role[];
```

Comment:
We prefer not to future proof. If only a single role is allowed, no need to use an array. Also, I was wondering - I believe the most common use case is to use a single role assumable by ec2.amazonaws.com. Does it make sense to make this the default? So if I don't specify a role, one can be created for me.

Reply:
@eladb 
_**We prefer not to future proof. If only a single role is allowed, no need to use an array.**_ 

I was trying to follow what is defined in CloudFormation.  It's defined as an array there.  Let me rework it.

_**Also, I was wondering - I believe the most common use case is to use a single role assumable by ec2.amazonaws.com. Does it make sense to make this the default? So if I don't specify a role, one can be created for me.**_

I think you are right about ec2 being the most common role, I will default it.

Reply:
> I was trying to follow what is defined in CloudFormation. It's defined as an array there. Let me rework it.

Yap. Totally get it. However, since the CDK is a client library, we have the wonderful privilege of semantic versioning. This means that we can design the type system to accurately describe the service capabilities, and evolve them as needed. If needed, we can even make breaking changes and reflect that in a major version bump. 

P.S. in this case, even if we at some point the instance profile can use multiple roles, it might be the right thing to simply add a property `additionalRoles` or even `roles` without breaking the original `role`. Anyway - that's a hypothetical discussion for the future 😉